src/proxy-animation.js

```diff
@@ -30,151 +30,470 @@ function createReadyPromise(details) {
         // Reject with a non-visible AbortError.
         reject(new DOMException("Animation aborted", "AbortError"));
       } else {
+        // Apply pending playbackRate
+        if (typeof details.pendingPlaybackRate == 'number') {
+          const previousCurrentTime = details.animation.currentTime;
+          details.animation.playbackRate = details.pendingPlaybackRate;
+          const timelineTime = details.timeline.currentTime;
+          details.startTime = details.pendingPlaybackRate ?
+              timelineTime - previousCurrentTime / details.pendingPlaybackRate :
+              previousCurrentTime;
+          details.pendingPlaybackRate = null;
+        }
+        switch(details.playState) {
+          case 'paused':
+            details.startTime = null;
+            break;
+
+          case 'running':
+          case 'finished':
+            details.holdTime = null;
+        }
         resolve();
       }
     });
   });
 }
 
+function getAnimationProperty(details, name) {
+  if (details.timeline && (name in details))
```

Comment:
I'm a bit concerned about sometimes returning the property from details and sometimes from the animation. I think we should remove the name in details check as it should be constant for a given property whether it is stored in the proxy details or in the base animation.

Reply:
start and hold time are examples of places where they will differ.  The underlying animation is locked in a pause pending state, so it will have a hold time but no start time.  Having said, that,  I can remove this method to make it clear where we are fetching the property from.  My reason for adding in the first place, was to cut down on repetitious code.  Having said that, possibly better to be clear.